src/units/abilities.cpp

```diff
@@ -854,6 +854,41 @@ std::string attack_type::weapon_specials(bool only_active, bool is_backstab) con
 		}
 	}
 
+	assert(display::get_singleton());
+	const unit_map& units = display::get_singleton()->get_units();
+	if(self_){
+		std::set<std::string> checking_name;
+		for (const config::any_child sp : (*self_).abilities().all_children_range()){
+			const bool active = check_self_abilities_impl(shared_from_this(), other_attack_, sp.cfg, self_, self_loc_, AFFECT_EITHER, sp.key);
+
+			const std::string& name = active ? sp.cfg["name"].str() : "";
+
+			if (!name.empty() && checking_name.count(name) == 0) {
+				checking_name.insert(name);
+				if (!res.empty()) res += ", ";
```

Comment:
I'd prefer this be on separate lines with braces.

Reply:
Done